Libraries/libutil/Headers/libutil/MemoryProcessContext.h

```diff
@@ -0,0 +1,89 @@
+/**
+ Copyright (c) 2015-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#ifndef __libutil_MemoryProcessContext_h
+#define __libutil_MemoryProcessContext_h
+
+#include <libutil/ProcessContext.h>
+
+namespace libutil {
+
+/*
+ * A process context with arbitrary values.
+ */
+class MemoryProcessContext : public ProcessContext {
+private:
+    std::string _executablePath;
+    std::string _currentDirectory;
+
+private:
+    std::unordered_map<std::string, std::string> _environmentVariables;
+
+private:
+    int32_t     _userID;
+    int32_t     _groupID;
+    std::string _userName;
+    std::string _groupName;
+
+public:
+    MemoryProcessContext(
+        std::string const &executablePath,
+        std::string const &currentDirectory,
+        std::unordered_map<std::string, std::string> const &environmentVariables,
+        int32_t userID,
+        int32_t groupID,
+        std::string const &userName,
+        std::string const &groupName);
+    explicit MemoryProcessContext(ProcessContext const &processContext);
+    virtual ~MemoryProcessContext();
+
+public:
+    virtual std::string executablePath() const
+    { return _executablePath; }
+    std::string &executablePath()
```

Comment:
The mutable references here are mostly to be setters. The getters above should be `const &`, but that was harder to implement for `DefaultProcessContext` and has to be the same for both.